datalad/support/annexrepo.py

```diff
@@ -597,12 +597,38 @@ def get_size_from_key(key):
         """A little helper to obtain size encoded in a key"""
         if not key:
             return None
-        try:
-            size_str = key.split('-', 2)[1].lstrip('s')
-        except IndexError:
-            # has no 2nd field in the key
+
+        # TODO: this method can be more compact. we don't need particularly
+        #       elaborated error distinction
+
+        # see: https://git-annex.branchable.com/internals/key_format/
+        key_parts = key.split('--')
+        key_fields = key_parts[0].split('-')
+
+        s = S = C = None
+
+        for field in key_fields[1:]:  # first one has to be backend -> ignore
+            if field.startswith('s'):
+                # size of the annexed file content:
+                s = int(field[1:]) if field[1:].isdigit() else None
+            elif field.startswith('S'):
+                # we have a chunk and that's the chunksize:
+                S = int(field[1:]) if field[1:].isdigit() else None
+            elif field.startswith('C'):
+                # we have a chunk, this is it's number:
+                C = int(field[1:]) if field[1:].isdigit() else None
```

Comment:
am I missing some difference or all 3 lines are identical and just assign to different variables? if there is no difference, then I would have used something like 
```python
if field and field[0] in 'sSC':
    value = int(field[1:]) if field[1:].isdigit() else None
    if field[0] == 's':
        s = value
    elif ...
```
to address code duplication.  or might  have even got brave for something like

```python
if field and field[0] in 'sSC':
    locals().update({field[0]:  int(field[1:]) if field[1:].isdigit() else None})
```
;-)

Reply:
Right! And that's exactly what the mysterious comment was about. The case distinction is needed only if we want to be more specific in "parsing" errors. Since we don't (I think) in can be "compacted" this way. Doing it now.

Reply:
FTR: `locals()` can't be updated. So, changed to a dict comprehension but updating local variables "manually".

Reply:
> FTR: `locals()` can't be updated.

d'oh. indeed. why did I think they could?

<details>
<summary>demo</summary> 

```shell
$> cat /tmp/11.py

def func():
    c = 1
    locals().update({'c': 2})
    print("c=%s" % c)

func()

$> python /tmp/11.py
c=1

```
</details>